lib/Dialect/TritonGPU/Transforms/Utility.cpp

```diff
@@ -818,6 +819,24 @@ bool isPureUnaryInlineAsm(Operation *op) {
          inlineAsmOp.getPure();
 }
 
+int getComputeCapability(Operation *module) {
+  assert(module->hasAttr(triton::AttrTargetName) &&
+         "Expected a target attribute on the module operation");
+
+  StringAttr targetAttr =
+      cast<StringAttr>(module->getAttr(triton::AttrTargetName));
+
+  auto ref = targetAttr.strref();
+  assert(ref.starts_with("cuda:") &&
+         "expected target attr to be prefixed with \"cuda:\"");
+
+  auto capabilityStr = ref.drop_front(5); // drop the "cuda:"
+  int computeCapability = 80;             // default capability
+  assert(!capabilityStr.getAsInteger(10, computeCapability));
```

Comment:
I'm not feeling too strong about this, we can keep as is for upstream review if you prefer.

Reply:
The assert should have an associates message

Reply:
```
capabilityStr.getAsInteger(10, computeCapability) 
```
Should not be enclosed into an assert because is possible to compile out assertion and that code is necessary to initialize `computeCapability` to the value in the module.  Suggesting to place that line outside of the assert and in the assert check the returned value.

Reply:
thanks, that might have been hard to debug